src/Plutus/Model/Contract.hs

```diff
@@ -272,16 +275,22 @@ spend pkh val = do
   mSp <- spend' pkh val
   pure $ fromJust mSp
```

Comment:
I know it's not a part of this PR, but this is annoying, can we change `fromJust` to `fromMaybe (error "meaningful explanation")`

Reply:
Yes please

Reply:
Reported in a separate issue https://github.com/mlabs-haskell/plutus-simple-model/issues/68

Reply:
MonadFail